### PR TITLE
Add test suites

### PR DESCRIPTION
**Pull request by archanaserver**
This PR adds a test suite for the `hammer_cli_foreman_tasks` plugin in Hammer CLI Foreman.

### CODE REVIEW

**Comment by archanaserver** [comment]
@ofedoren This is now ready for the review. Would you mind looking into it?

**Comment by ekohl** [comment]
Note that currently it doesn't run CI, so we need to fix that either in Jenkins or with GHA.

File: lib/hammer_cli_foreman_tasks/async_command.rb [hunk 1]
```diff
@@ -1,3 +1,5 @@
+require 'hammer_cli_foreman'
```
Comment (ofedoren) [inline lib/hammer_cli_foreman_tasks/async_command.rb:1]: This should not be necessary...
```suggestion
```

File: test/functional/task_test.rb [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        skipped: [task]
+      })
+
+      result = run_cmd(cmd)
+      assert_equal(70, result.exit_code)  # Check the exit code
+    end
+  end
+
+  describe 'progress' do
+    let(:cmd) { base_cmd + ['progress', '--option-id', '1'] }
```
Comment (ofedoren) [inline test/functional/task_test.rb:88]:
```suggestion
    let(:cmd) { base_cmd + ['progress', '--id', '1'] }
```

File: test/functional/task_test.rb [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        total_resumed: 3,
+        total_failed: 1,
+        total_skipped: 1,
+        resumed: [task, task, task],
+        failed: [task],
+        skipped: [task]
+      })
+
+      result = run_cmd(cmd)
+      assert_equal(70, result.exit_code)  # Check the exit code
```
Comment (ofedoren) [inline test/functional/task_test.rb:83]:
```suggestion
      assert_equal(HammerCLI::EX_OK, result.exit_code)  # Check the exit code
```

Every (almost) command should return `0` or `HammerCLI::EX_OK` if everything is OK.

**Review by ofedoren — CHANGES_REQUESTED** [review]
Thanks, @archanaserver, and so sorry for making you wait for so long. Unfortunately, it fails locally:
```bash
/home/vagrant/.rbenv/versions/3.0.4/bin/ruby -I"lib:test:lib" -I"/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rake-10.1.1/lib" "/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rake-10.1.1/lib/rake/rake_test_loader.rb" "test/functional/task_test.rb" 
No permissions to create log dir /var/log/hammer.
File /var/log/hammer/hammer.log not writeable, won't log anything to the file!
Run options: --seed 2448

# Running tests:

FFFF

Finished tests in 0.126794s, 31.5472 tests/s, 47.3208 assertions/s.

  1) Failure:
task::info#test_0001_should show a task [/home/vagrant/hammer-cli-foreman-tasks/test/functional/task_test.rb:64]:
--- expected
+++ actual
@@ -1,8 +1,10 @@
 "ID:         1
-Action:     test_action
+Action:      
 State:      completed
 Result:     success
 Started at: 2023/10/31 00:00:00
 Ended at:   2023/10/31 00:00:00
 Duration:   1h 30m
-Owner:      admin"
+Owner:      admin
+
+"


  2) Failure:
task::list#test_0001_should list all tasks [/home/vagrant/hammer-cli-foreman-tasks/test/functional/task_test.rb:39]:
Regex didn't match the output.
Expected regex:
[ ]*ID[ ]*\|[ ]*ACTION[ ]*\|[ ]*STATE[ ]*\|[ ]*RESULT[ ]*\|[ ]*STARTED\ AT[ ]*\|[ ]*ENDED\ AT[ ]*\|[ ]*DURATION[ ]*\|[ ]*OWNER[ ]*\|[ ]*TASK\ ERRORS[ ]*
Expected fields:
ID | ACTION | STATE | RESULT | STARTED AT | ENDED AT | DURATION | OWNER | TASK ERRORS
Actual output:
---|--------|-----------|---------|---------------------|---------------------|----------|------
ID | ACTION | STATE     | RESULT  | STARTED AT          | ENDED AT            | DURATION | OWNER
---|--------|-----------|---------|---------------------|---------------------|----------|------
1  |        | completed | success | 2023/10/31 00:00:00 | 2023/10/31 00:00:00 | 1h 30m   | admin
---|--------|-----------|---------|---------------------|---------------------|----------|------


  3) Failure:
task::progress#test_0001_should show the progress of the task [/home/vagrant/hammer-cli-foreman-tasks/test/functional/task_test.rb:107]:
--- expected
+++ actual
@@ -1 +1,4 @@
-""
+"Error: Unrecognised option '--option-id'.
+
+See: 'hammer task progress --help'.
+"


  4) Failure:
task::resume#test_0001_should resume tasks paused in error state [/home/vagrant/hammer-cli-foreman-tasks/test/functional/task_test.rb:83]:
Expected: 70
  Actual: 0

4 tests, 6 assertions, 4 failures, 0 errors, 0 skips
rake aborted!
Command failed with status (1): [ruby -I"lib:test:lib" -I"/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rake-10.1.1/lib" "/home/vagrant/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/rake-10.1.1/lib/rake/rake_test_loader.rb" "test/functional/task_test.rb" ]

Tasks: TOP => test
(See full trace by running task with --trace)
```

But at least we have something to test, that's great :) Few suggestions as of now:

**Pull request closed without merging**